code/modules/surgery/organs/blood.dm

```diff
@@ -72,11 +72,17 @@
 
 		bleed_rate = max(bleed_rate - 0.5, temp_bleed)//if no wounds, other bleed effects (heparin) naturally decreases
 
-		if(internal_bleeding_rate && !(status_flags & FAKEDEATH))
-			bleed_internal(internal_bleeding_rate)
+		if(internal_bleeding_rate && !(status_flags & FAKEDEATH) && !reagents.has_reagent("quikclot"))
+			if(reagents.has_reagent("tran_acid"))
+				bleed_internal(internal_bleeding_rate * 0.3) 	//bleed_internal(internal_bleeding_rate - 0.7 * reagents.has_reagent("tran_acid") * internal_bleeding_rate) efficient alternative 
+			else
+				bleed_internal(internal_bleeding_rate)
```

Comment:
Move this whole "bleed rate is modified by this percentage when you have reagents in you" logic into a standalone proc (`get_bleed_modifier()` or something). You'll then be able to plug it in both for internal and external bleeding without the need to duplicate code.